lib/config.js

```diff
@@ -220,13 +238,8 @@ function getLocalConfig(thisConfig, directory) {
         debug("Loading " + localConfigFile);
         localConfig = loadConfig(localConfigFile);
 
-        if (path.basename(localConfigFile) !== LOCAL_CONFIG_FILENAME) {
-
-            // Don't consider a local config file found if the package.json doesn't have the eslintConfig field.
-            if (!localConfig.hasOwnProperty(PACKAGE_CONFIG_FIELD_NAME)) {
-                continue;
-            }
-            localConfig = localConfig[PACKAGE_CONFIG_FIELD_NAME] || {};
+        if (!Object.keys(localConfig).length) {
```

Comment:
Not sure I understand the point of this. An empty config should have no effect, so if just let it through.